Add progress bar sync

robertDurst opened the pull request:
Ported the progress bar for chain sync to v2. It updates user w/ the current block height. Once the chain is synced, the progress bar disappears.

Kept with the current code style with inline css.

Closes #188 

## Overall View
<img width="1440" alt="screen shot 2018-02-04 at 5 41 05 pm" src="https://user-images.githubusercontent.com/16689974/35785213-afc2185e-09d3-11e8-9b22-07457c7dff0d.png">

## Close up
<img width="172" alt="screen shot 2018-02-04 at 5 41 10 pm" src="https://user-images.githubusercontent.com/16689974/35785227-bb57bc78-09d3-11e8-9c49-30fbbe7e1668.png">

## GIF w/ updating block height and v1 animation
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/16689974/35785289-3867add6-09d4-11e8-8fc4-0093fba4201e.gif)

bryanvu commented:
Hey @robertDurst, the previous PR for this, by @jkdufair, was waiting on an lnd PR. Is this one dependent on the same PR?

robertDurst commented:
@bryanvu I saw his. This one is specific to porting the progress bar over to V2 and it does not have the cool sync like @jkdufair. It also is not dependent on the same PR -- it doesn't show the progress per say, it just displays current synced block height.

tanx commented on src/views/sidebar.js:
```diff
@@ -1,14 +1,27 @@
 class Sidebar extends Component {
+  componentDidMount() {
+    interval = setInterval(() => {
+      actionsInfo.getInfo();
```
We already call `getInfo()` in the constructor in `src/actions/info.js`. No need to call it here again. So it looks like `componentDidMount` and `componentWillUnmount` are not needed.

tanx commented on src/views/sidebar.js:
```diff
@@ -70,6 +83,42 @@ class Sidebar extends Component {
+            className="syncing"
+          >
+            <style
+              dangerouslySetInnerHTML={{
```
Do we really need/want to `dangerouslySetInnerHTML` ?

robertDurst replied:
This was copied straight from v1. On second thought, is not really necessary. Will fix.

robertDurst commented on src/actions/info.js:
```diff
@@ -16,6 +16,10 @@ class ActionsInfo {
       this._store.pubKey = response.identity_pubkey;
       this._store.syncedToChain = response.synced_to_chain;
       this._store.blockHeight = response.block_height;
+      if (!response.synced_to_chain) {
```
No longer call ComponentDidMount in sidebar.js. Added an if statement to getInfo that will continuously call getInfo until sync is complete. This allows for the updating of blockHeight and syncedToChain and rerendering of the sidebar to show current status.

tanx replied:
Great. Any reason you chose 1000ms and not the `RETRY_DELAY` as used in the catch block?

robertDurst replied:
No, not really, I can change it.

tanx commented on src/views/sidebar.js:
```diff
@@ -45,6 +35,13 @@ class Sidebar extends Component {
+                    50% { opacity: 0.8 }
+                    100% { opacity: 1 }
+                  }`;
+    styleSheet.insertRule(keyframes, styleSheet.cssRules.length);
```
Awesome 👍

tanx commented:
@robertDurst thanks for the updates. Looks good in terms of the code. Only left to update the unit tests: https://github.com/lightninglabs/lightning-app/blob/v2/test/unit/actions/info.spec.js#L27-L43

robertDurst commented:
@tanx 👍

robertDurst commented on test/unit/actions/info.spec.js:
```diff
@@ -25,12 +25,34 @@ describe('Actions Info Unit Tests', () => {
+    it('should get public key, synced to chain, and block height', async () => {
       actionsGrpc.sendCommand.withArgs('getInfo').resolves({
         identity_pubkey: 'some-pubkey',
+        synced_to_chain: 'true/false',
```
Expanded basic test to include synced to chain and block height

robertDurst commented:
I added the following for tests:

1) Expanded basic test to include synced to chain and block height.
2) Included two tests to test the synced progress part of getInfo I added
a. Check that if chain is synced getInfo only runs once
b. Check that if the chain is not synced, getInfo is repeated
